diff --git a/node_modules/react-native-autogrow-textinput/android/src/main/java/com/wix/autogrowtextinput/AutoGrowTextInputModule.java b/node_modules/react-native-autogrow-textinput/android/src/main/java/com/wix/autogrowtextinput/AutoGrowTextInputModule.java
index 07c76d6..b2d1549 100644
--- a/node_modules/react-native-autogrow-textinput/android/src/main/java/com/wix/autogrowtextinput/AutoGrowTextInputModule.java
+++ b/node_modules/react-native-autogrow-textinput/android/src/main/java/com/wix/autogrowtextinput/AutoGrowTextInputModule.java
@@ -52,7 +52,7 @@ public class AutoGrowTextInputModule extends ReactContextBaseJavaModule {
                 if (param.hasKey("maxHeight") && !param.isNull("maxHeight")) {
                     mMaxHeight = dpToPx(param.getDouble("maxHeight"));
                 }
-                editText.setBlurOnSubmit(false);
+                editText.setSubmitBehavior("submit");
                 editText.setOnTouchListener(new View.OnTouchListener() {
                     @Override
                     public boolean onTouch(View v, MotionEvent event) {
diff --git a/node_modules/react-native-autogrow-textinput/src/AutoGrowingTextInput.js b/node_modules/react-native-autogrow-textinput/src/AutoGrowingTextInput.js
index 68531dc..fc9be73 100644
--- a/node_modules/react-native-autogrow-textinput/src/AutoGrowingTextInput.js
+++ b/node_modules/react-native-autogrow-textinput/src/AutoGrowingTextInput.js
@@ -1,6 +1,7 @@
 import React, {Component} from 'react';
 import ReactNative, {TextInput, Platform, NativeModules} from 'react-native';
 import PropTypes from 'prop-types';
+import PasteInput from 'react-native-paste-image-input';
 
 const ANDROID_PLATFORM = (Platform.OS === 'android');
 
@@ -45,7 +46,7 @@ export default class AutoGrowingTextInput extends Component {
 
   render() {
     return (
-      <TextInput
+      <PasteInput
         multiline
         {...this.props} {...this.style}
         style={[this.props.style, {height: 'auto'}]}
